src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMKeyValue.java

```diff
@@ -264,6 +264,22 @@ public boolean equals(Object obj) {
                 return false;
             }
         }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            try {
+                if (getPublicKey() != null) {
+                    RSAPublicKey rsaKey = (RSAPublicKey)getPublicKey();
+                    result = 31 * result + rsaKey.getPublicExponent().hashCode();
+                    result = 31 * result + rsaKey.getModulus().hashCode();
+                }
+            } catch (KeyException ke) {
+                // no key available
```

Comment:
I would probably return `super.hashCode(`) here and also change line 264 to return `super.equals()` so if the key can't be extracted only an exact match of the same object will equal.

We should probably add equals/hashCode methods for the other *KeyValue subclasses, though we could do that in another PR. Note that the XMLDSig API does not require implementations to override equals/hashCode. However, it is useful for testing purposes.